Common/src/CConfig.cpp

```diff
@@ -4314,6 +4364,702 @@ void CConfig::SetPostprocessing(unsigned short val_software, unsigned short val_
+      Enthalpy_Formation[0] = 0.0;          //N2
+      // Reference temperature (JANAF values, [K])
+      Ref_Temperature[0] = 0.0;
+
+      /*        nElStates[0] = 0;
+           CharElTemp   = new double *[nSpecies];
+           degen        = new double *[nSpecies];
+
+           OSPthetae    = new double[nElStates[0]];
+           OSPthetae[0] = 1.0;
+           OSPg         = new double[nElStates[0]];
+           OSPg[0]      = 1.0;
+
+           CharElTemp[0] = OSPthetae;
+           degen[0] = OSPg;*/
+
+      break;
+
+    case N2:
+
+      /*--- Check for errors in the initialization ---*/
+      init_err = false;
+      if (nSpecies != 2) {
+        cout << "CONFIG ERROR: nSpecies mismatch between gas model & gas composition" << endl;
+        init_err = true;
+      }
+      mf = 0.0;
+      for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+        mf += Gas_Composition[iSpecies];
+      if (mf != 1.0) {
+        cout << "CONFIG ERROR: Intial gas mass fractions do not sum to 1!" << endl;
+        init_err = true;
+      }
+
+      /*--- Define parameters of the gas model ---*/
+      nReactions  = 2;
+      ionization  = false;
+
+      /*--- Allocate vectors for gas properties ---*/
+      Wall_Catalycity      = new su2double[nSpecies];
+      Molar_Mass           = new su2double[nSpecies];
+      CharVibTemp          = new su2double[nSpecies];
+      RotationModes        = new su2double[nSpecies];
+      Enthalpy_Formation   = new su2double[nSpecies];
+      Ref_Temperature      = new su2double[nSpecies];
+      Diss                 = new su2double[nSpecies];
+      ArrheniusCoefficient = new su2double[nReactions];
+      ArrheniusEta         = new su2double[nReactions];
+      ArrheniusTheta       = new su2double[nReactions];
+      Tcf_a                = new su2double[nReactions];
+      Tcf_b                = new su2double[nReactions];
+      Tcb_a                = new su2double[nReactions];
+      Tcb_b                = new su2double[nReactions];
+      nElStates            = new unsigned short[nSpecies];
+      Reactions = new int**[nReactions];
+      for (unsigned short iRxn = 0; iRxn < nReactions; iRxn++) {
+        Reactions[iRxn] = new int*[2];
+        for (unsigned short ii = 0; ii < 2; ii++)
+          Reactions[iRxn][ii] = new int[6];
+      }
+
+      Blottner  = new su2double*[nSpecies];
+      for (iSpecies = 0; iSpecies < nSpecies; iSpecies++)
+        Blottner[iSpecies] = new su2double[3];
+
+      // Omega[iSpecies][jSpecies][iCoeff]
+      Omega00 = new su2double**[nSpecies];
+      Omega11 = new su2double**[nSpecies];
+      for (iSpecies = 0; iSpecies < nSpecies; iSpecies++) {
+        Omega00[iSpecies] = new su2double*[nSpecies];
+        Omega11[iSpecies] = new su2double*[nSpecies];
+        for (jSpecies = 0; jSpecies < nSpecies; jSpecies++) {
+          Omega00[iSpecies][jSpecies] = new su2double[4];
+          Omega11[iSpecies][jSpecies] = new su2double[4];
+        }
+      }
```

Comment:
These variables should definitely use a "3D" storage type, this triple nesting of pointers is not good for performance and clutters the code a lot.